### ISSUE TITLE
AWS ECS Fargate throws "Unable to lookup role in URI"

### ISSUE THREAD

**Issue by jpadilladev** [hazelcast/hazelcast-aws#174]
Hi,

I am trying to run a Spring Boot 2 application on AWS, using ECS along with Fargate. I´ve been following the readme file, and added the default configuration so it uses the Task Role with the specified permissions.

This is my Java configuration
```
config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
			config.getNetworkConfig().getInterfaces().setEnabled(true).addInterface("10.0.*.*");
			config.getNetworkConfig().getJoin()
					.getAwsConfig()
					.setEnabled(true);
```

When deployed to AWS, it throws the following exception:

```
IOException: Unable to lookup role in URI `http://169.254.169.254/latest/meta-data/placement/availability-zone/
```

Is this because the Fargate Metadata endpoint may be different? See [this ](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint.html)

Thank you

**Comment by leszko** [comment]
Do you build `hazelcast-aws` dependency by yourself? The Fargate support was not released yet. It will be released in `hazelcast-aws` version [3.2](https://github.com/hazelcast/hazelcast-aws/milestone/10).

**Comment by jpadilladev** [comment]
I thought v3.1 had ECS Fargate support already, it was explained on the readme...

Anyway, I built locally v3.2 and deployed it, and it works smoothly now.

Is it v3.2 to be released anytime soon? 

Thanks

**Comment by leszko** [comment]
Yes, it will be released in a week or two.

**Comment by tulans** [comment]
which configuration needs to be part of cluster.xml to enable ECS Fargate support ?. 

<aws enabled="false">
                <iam-role>Pro...</iam-role>
                <region>ap-south-1</region>
                <security-group-name>m_cmn_ap</security-group-name>
                <tag-key>App</tag-key>
                <tag-value>m_cmn_common</tag-value>
</aws>
or 
<discovery-strategies>
                <discovery-strategy
                        enabled="true" class="com.hazelcast.aws.AwsDiscoveryStrategy">
                    <properties>
                        <property name="iam-role">prod-ecsrsettings</property>
                        <property name="region">ap-sh-1</property>
                        <property name="tag-key">SERVICE_NAME</property>
                        <property name="tag-value">usettings</property>
                    </properties>
                </discovery-strategy>
</discovery-strategies>

It would really help if someone can help publish a blog on hazelcast clustering with AWS ECS fargate.

**Comment by pivovarit** [comment]
Fargate support will be properly covered once released

**Comment by jpadilladev** [comment]
@tulans Using the default configuration on the readme will use the task role and will search for cluster members on the current aws cluster.
It worked for me just with the following (and some IAM role privileges tunning)

```
config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
			config.getNetworkConfig().getInterfaces().setEnabled(true).addInterface("10.0.*.*");
			config.getNetworkConfig().getJoin()
					.getAwsConfig()
					.setEnabled(true);
``` 

This is solved with v3.2, so I will close the issue.
Thanks.
